src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -2898,6 +3087,19 @@ MillisecondsBetweenTimestamps(instr_time startTime, instr_time endTime)
 }
 
 
+/*
+ * MicrosecondsBetweenTimestamps is a helper to get the number of microseconds
+ * between timestamps when it is expected to be small enough to fit in a
+ * long.
+ */
```

Comment:
> when it is expected to be small enough to fit in a long.

If we use `uint64` as the return time like I suggested below, then this is 500k years. If we do that change, I don't think it's necessary to say that we expect it to fit in that.

If we keep a long, I think it's probably good to qualify how  long is "small enough to fit in a long". This is ~6 years.

Reply:
I changed it to uint64 still I don't expect a query execution to take 6 years :D

Reply:
Can you remove this part of the comment now:
>  when it is expected to be small enough to fit in a long.

It's definitely not necessary to warn people against it not fitting in a uint64 now.